### PR TITLE
Make all commercial modules return promises in all situations

### PR DESCRIPTION
**Pull request by regiskuckaertz**
This addresses an issue where commercial.js [expects](https://github.com/guardian/frontend/blob/master/static/src/javascripts/bootstraps/commercial.js#L97) all commercial modules to return promises in order to track module performance.

cc @guardian/commercial-dev @piuccio

### CODE REVIEW

File: static/src/javascripts/projects/common/modules/commercial/hosted-video.js [hunk 1]
```diff
@@ -85,99 +85,103 @@ define([
...
+                        if ($hostedNext.length && ab.getParticipations().HostedAutoplay
+                            && (ab.getParticipations().HostedAutoplay.variant === 'variant1' || ab.getParticipations().HostedAutoplay.variant === 'variant2')) {
+                            if (ab.getParticipations().HostedAutoplay.variant === 'variant2') {
+                                fastdom.write(function () {
+                                    $hostedNext.addClass('hosted-next-autoplay--variant2');
+                                });
+                            }
                             fastdom.write(function () {
-                                $hostedNext.addClass('hosted-next-autoplay--variant2');
+                                $('.js-hosted-fading').addClass('hosted-autoplay-ab');
```
Comment (regiskuckaertz) [inline static/src/javascripts/projects/common/modules/commercial/hosted-video.js]: Huh, I did not do that. Pulling latest master to make sure this is correct.

File: static/src/javascripts/projects/common/modules/commercial/hosted-video.js [hunk 2]
```diff
@@ -85,99 +85,103 @@ define([
     }
 
     function init() {
-        require(['bootstraps/enhanced/media/main'], function () {
```
Comment (rich-nguyen) [inline static/src/javascripts/projects/common/modules/commercial/hosted-video.js]: Ditto https://github.com/guardian/frontend/pull/13623

Comment (rich-nguyen) [inline static/src/javascripts/projects/common/modules/commercial/hosted-video.js]: It was suspected that this promise was causing hosted problems, and was removed in https://github.com/guardian/frontend/pull/13616, so I've reinstated it too.

**Comment by rich-nguyen** [comment]
Whoa I missed loads of them. Thanks @regiskuckaertz

**Comment by piuccio** [comment]
👍

**Comment by rich-nguyen** [comment]
looks good 👍

**Comment by regiskuckaertz** [comment]
@rich-nguyen no worries 😅

File: static/src/javascripts/projects/common/modules/commercial/hosted-video.js (hunk 2 above)
Comment (regiskuckaertz) [inline static/src/javascripts/projects/common/modules/commercial/hosted-video.js]: Oh right, ok who goes first? Just make sure you replace the `return` statement with a call to `resolve()` instead

Comment (regiskuckaertz) [inline static/src/javascripts/projects/common/modules/commercial/hosted-video.js]: Let's roll yours first.

**Pull request merged**